src/main/java/org/ballerinalang/containers/docker/impl/DefaultBallerinaDockerClient.java

```diff
@@ -152,11 +152,16 @@ public boolean deleteImage(String packageName, String dockerEnv, String imageNam
     public String getImage(String imageName, String dockerEnv) {
         DockerClient client = getDockerClient(dockerEnv);
         List<Image> images = client.image().list().filter(imageName).endImages();
-        for (Image image : images) {
-            String currentImageName = image.getRepoTags().get(0);
-            if (currentImageName.equals(imageName + ":" + Constants.IMAGE_VERSION_LATEST) ||
-                    currentImageName.equals(imageName)) {
-                return currentImageName;
+
+        if (images != null && images.size() > 0) {
+            for (Image image : images) {
+                if (image.getRepoTags() != null) {
+                    String currentImageName = image.getRepoTags().get(0);
```

Comment:
@lankavitharana Thanks for the input! I've done the changes as suggested in the commit 	eb54dc9. Please check.